src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -0,0 +1,250 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Retargeting;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    public class ReadOnlyStructsTests : CompilingTestBase
+    {
+        [Fact()]
+        public void WriteableInstanceAutoPropsInRoStructs()
+        {
+            var text = @"
+public readonly struct A
+{
+    // ok
+    int ro => 5;
```

Comment:
Yes. the backing field is readonly. 
And even if it was not, and if we allowed autoproperties to write through `this` here - it would be strange that autoproperties can do that while no other regular member can. 

---
In reply to: [121437834](https://github.com/dotnet/roslyn/pull/20142#discussion_r121437834) [](ancestors = 121437834)